crates/worker/src/partition/mod.rs

```diff
@@ -517,6 +560,224 @@ where
         Ok(())
     }
 
+    // --- RPC Handling
+
+    async fn on_rpc(
+        &mut self,
+        rpc: Incoming<(
+            PartitionProcessorRpcRequestId,
+            PartitionProcessorRpcRequestInner,
+        )>,
+        partition_store: &mut PartitionStore,
+    ) {
+        let sender_node_id = *rpc.peer();
+        let (response_tx, (request_id, body)) = rpc.split();
+        match body {
+            PartitionProcessorRpcRequestInner::SubmitInvocation(
+                service_invocation,
+                SubmitInvocationReplyOn::Appended,
+            ) => {
+                let res = match self
+                    .leadership_state
+                    .self_propose(
+                        service_invocation.partition_key(),
+                        Command::Invoke(service_invocation),
+                    )
+                    .await
+                {
+                    Ok(()) => Ok(PartitionProcessorRpcResponse::Appended),
```

Comment:
Once `self_propose` completes it is not guaranteed that the record was appended to Bifrost.

Reply:
how do i guarantee that? I guess I need to stop using the bifrost background appender or what?

Reply:
You can call `enqueue_with_notification` which gives you a future to await on. However, this should not happen as part of this method since it would block the `PartitionProcessor` from processing other requests.

Reply:
Done